libsolidity/ast/Types.cpp

```diff
@@ -2632,11 +2632,13 @@ FunctionType::FunctionType(FunctionDefinition const& _function, Kind _kind):
 
 	for (ASTPointer<VariableDeclaration> const& var: _function.parameters())
 	{
+		solAssert(var->annotation().type, "Parameter type is not yet available in the AST.");
```

Comment:
For what function parameter types is this assertion expected to fail?

Reply:
It will fail if `type` is null, which may be the case if you forget to run `DeclarationTypeChecker` during analysis. Without this assertion you will just get a memory violation at a random place and only if you happen to use the annotation. I had a really hard time debugging this and such an assertion would have saved me a lot of time.

It's not all that relevant to this PR anymore because I switched to creating the whole `CompilerStack` in tests (rather than running specific analysis phases manually). There are a few commits with small tweaks like this one in this PR. I could extract them into a separate PR but I wasn't sure if it's worth it since these tweaks are so small (some of them are just typo fixes).